augur/merge.py

```diff
@@ -208,125 +440,204 @@ def run(args):
-        # allowed inside f-string expressions, hence the *newline* variable.
-        newline = '\n'
-        query = dedent(f"""\
-            select
-                {(',' + newline + '                ').join(select_list)}
-            from
-                {(newline + '                ').join(from_list)}
-            ;
-            """)
-
-
-        # Write merged metadata as export from SQLite db.
-        #
-        # Assume TSV like nearly all other extant --output-metadata options.
-        print_info(f"Merging metadata and writing to {args.output_metadata!r}…")
-        print_debug(query)
-        sqlite3(db_path,
-            f'.mode csv',
-            f'.separator "\\t" "\\n"',
-            f'.headers on',
-            f'.once {sqlite_quote_dot(f"|{augur} write-file {shquote(args.output_metadata)}")}',
-            query)
-
-    except SQLiteError as err:
-        delete_db = False
-        raise AugurError(str(err)) from err
-
-    finally:
-        if delete_db:
-            os.unlink(db_path)
+    for name, path in sequences:
+        if name == "":
+            yield UnnamedSequenceFile(path)
+        else:
+            yield NamedSequenceFile(name, path)
+
+
+def load_sequences(db: Database, sequences: List[Union[NamedSequenceFile, UnnamedSequenceFile]]):
+    for s in sequences:
+        print_info(f"Reading sequence IDs from {s.path!r}…")
+        ids = [seq.id for seq in read_sequences(s.path)]
+
+        if duplicates := [item for item, count in count_unique(ids) if count > 1]:
+            raise AugurError(dedent(f"""\
+                IDs must be unique within a sequence input file.
+
+                The following {_n("entry is", "entries are", len(duplicates))} duplicated in {s.path!r}:
+
+                  {indented_list([repr(x) for x in duplicates], '                ' + '  ')}
+                """))
+
+        # FIXME: Skip for unnamed sequences? This is only used for named sequences to check against paired metadata.
+        db.run(f"create table {sqlite_quote_id(s.table_name)} ({sqlite_quote_id(SEQUENCE_ID_COLUMN)} text);")
+        values = ", ".join([f"('{id}')" for id in ids])
+        db.run(f"insert into {sqlite_quote_id(s.table_name)} ({sqlite_quote_id(SEQUENCE_ID_COLUMN)}) values {values};")
+
+        db.run(f'create unique index {sqlite_quote_id(f"{s.table_name}_id")} on {sqlite_quote_id(s.table_name)}({sqlite_quote_id(SEQUENCE_ID_COLUMN)});')
+
+
+def cat(filepath: str):
+    cat = "cat"
+
+    if filepath.endswith(".gz"):
+        cat = "gzcat"
+    if filepath.endswith(".xz"):
+        cat = "xzcat"
+    if filepath.endswith(".zst"):
+        cat = "zstdcat"
+
+    return cat, filepath
+
+
+def merge_sequences(
+        sequences: List[Union[NamedSequenceFile, UnnamedSequenceFile]],
+        output_sequences: str,
+    ):
+    # Confirm that seqkit is installed.
+    if which("seqkit") is None:
+        raise AugurError("'seqkit' is not installed! This is required to merge sequences.")
```

Comment:
This duplicates a check that's also done on L717; that latter check is also looking at the `SEQKIT` env var, which this check isn't. 

I'm not sure which one is correct, but it doesn't seem like both are useful?